federation/pkg/federation-controller/ingress/ingress_controller.go

```diff
@@ -747,24 +736,9 @@ func (ic *IngressController) reconcileIngress(ingress types.NamespacedName) {
 			return
 		}
 		desiredIngress := &extensionsv1beta1.Ingress{}
-		objMeta, err := api.Scheme.DeepCopy(&baseIngress.ObjectMeta)
-		if err != nil {
-			glog.Errorf("Error deep copying ObjectMeta: %v", err)
-		}
-		objSpec, err := api.Scheme.DeepCopy(&baseIngress.Spec)
-		if err != nil {
-			glog.Errorf("Error deep copying Spec: %v", err)
-		}
-		objMetaCopy, ok := objMeta.(*metav1.ObjectMeta)
-		if !ok {
-			glog.Errorf("Internal error: Failed to cast to *metav1.ObjectMeta: %v", objMeta)
-		}
-		desiredIngress.ObjectMeta = *objMetaCopy
-		objSpecCopy, ok := objSpec.(*extensionsv1beta1.IngressSpec)
-		if !ok {
-			glog.Errorf("Internal error: Failed to cast to extensionsv1beta1.Ingressespec: %v", objSpec)
-		}
-		desiredIngress.Spec = *objSpecCopy
+		desiredIngress.ObjectMeta = *baseIngress.ObjectMeta.DeepCopy()
+		baseIngress.Spec.DeepCopyInto(&desiredIngress.Spec)
```

Comment:
Why the assignment above but the DeepCopyInto here?  Can you help me to understand why they need different approaches?

Reply:
both are equivalent. The second line uses less allocations. Let me unify them.